scalding-quotation/src/main/scala/com/twitter/scalding/quotation/Projection.scala

```diff
@@ -8,6 +8,87 @@ case class TypeName(asString: String) extends AnyVal
 sealed trait Projection {
   def andThen(accessor: Accessor, typeName: TypeName): Projection =
     Property(this, accessor, typeName)
+
+  /**
+   * Used to check if a projection is based on a specific type.
+   * For instance, given the type name `Person`:
+   *
+   * `Person.name`   => true
+   * `Contact.phone` => false
+   */
+  def isBasedOntype(typeName: TypeName): Boolean = {
```

Comment:
nit: `type` should start with capital letter.

Personally I prefer to see rootProjection method here and just inline `rootProjection(this).typeName == typeName` at callsite.

Reply:
done